fix compilation error with Intel compilers
error: no instance of function template "boost::property_tree::xml_parser::xml_writer_make_settings" matches the argument list
author: Kenneth Hoste (HPC-UGent)
+++ bcl2fastq/src/cxx/lib/io/Xml.cpp	2017-12-05 21:18:28.565412877 +0100
@@ -169,17 +169,17 @@
     {
         unindex(*tree.begin(), treeWithIndexAttributes);
 #ifndef WIN32
-        boost::property_tree::write_xml(os, treeWithIndexAttributes, boost::property_tree::xml_writer_make_settings(' ', 2));
+        boost::property_tree::write_xml(os, treeWithIndexAttributes, boost::property_tree::xml_writer_make_settings<ptree::key_type>(' ', 2));
 #else
-        boost::property_tree::write_xml(os, treeWithIndexAttributes, boost::property_tree::xml_writer_make_settings<std::string>(' ', 2));
+        boost::property_tree::write_xml(os, treeWithIndexAttributes, boost::property_tree::xml_writer_make_settings<ptree::key_type>(' ', 2));
 #endif
     }
     else
     {
 #ifndef WIN32
-        boost::property_tree::write_xml(os, tree, boost::property_tree::xml_writer_make_settings(' ', 2));
+        boost::property_tree::write_xml(os, tree, boost::property_tree::xml_writer_make_settings<ptree::key_type>(' ', 2));
 #else
-        boost::property_tree::write_xml(os, tree, boost::property_tree::xml_writer_make_settings<std::string>(' ', 2));
+        boost::property_tree::write_xml(os, tree, boost::property_tree::xml_writer_make_settings<ptree::key_type>(' ', 2));
 #endif
     }
     return os;
